From 45335ddd43efed3feaf4c142b7a46dee9122cc90 Mon Sep 17 00:00:00 2001
From: Hanna Hawa <hannah@marvell.com>
Date: Tue, 12 Jul 2016 12:33:43 +0300
Subject: [PATCH 0342/1345] config: a70x0, a80x0: disable XOR entries from
 A70x0, A80x0 DTS

commit  7e231a8ec309b5fd5e0169ffe49554b712728825 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

XOR engine is not functional with Silicon ERRATA, according to ERRATA
the byte count not supporting all size options, and not supported
64k byte count, that's the buffer size used (PAGE_SIZE) in the kernel
in real environment.
This issue will be resolved in next Silicon revision, for now
this patch disable the XOR driver for Armada-70x0, and Armada-80x0

Change-Id: Ia1664a96e0f24a180e4a3d0d27a7b5e3ba841b75
Signed-off-by: Hanna Hawa <hannah@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/31063
Reviewed-by: Nadav Haklai <nadavh@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Neta Zur Hershkovits <neta@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-ap806.dtsi   |    4 ++++
 arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi |    2 ++
 arch/arm64/boot/dts/marvell/armada-cp110.dtsi   |    2 ++
 3 files changed, 8 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-ap806.dtsi b/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
index ed9011a..f452b90 100644
--- a/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
@@ -218,6 +218,7 @@
 				      <0x410000 0x1000>;
 				msi-parent = <&gic_v2m0>;
 				dma-coherent;
+				status = "disabled";
 			};
 
 			xor@420000 {
@@ -226,6 +227,7 @@
 				      <0x430000 0x1000>;
 				msi-parent = <&gic_v2m0>;
 				dma-coherent;
+				status = "disabled";
 			};
 
 			xor@440000 {
@@ -234,6 +236,7 @@
 				      <0x450000 0x1000>;
 				msi-parent = <&gic_v2m0>;
 				dma-coherent;
+				status = "disabled";
 			};
 
 			xor@460000 {
@@ -242,6 +245,7 @@
 				      <0x470000 0x1000>;
 				msi-parent = <&gic_v2m0>;
 				dma-coherent;
+				status = "disabled";
 			};
 
 			thermal@6f808c {
diff --git a/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi
index 20c0101..97ff53a 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi
@@ -129,6 +129,7 @@ cpxor0@6a0000 {
 	dma-coherent;
 	msi-parent = <&gic_v2m0>;
 	clocks = <&gateclk 14>, <&gateclk 8>;
+	status = "disabled";
 };
 
 cpxor1@6c0000 {
@@ -138,6 +139,7 @@ cpxor1@6c0000 {
 	dma-coherent;
 	msi-parent = <&gic_v2m0>;
 	clocks = <&gateclk 14>, <&gateclk 7>;
+	status = "disabled";
 };
 
 mdio@12a200 {
diff --git a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
index 28f1c19..eeb82f7 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
@@ -203,6 +203,7 @@ cpxor0@6a0000 {
 	dma-coherent;
 	msi-parent = <&gic_v2m0>;
 	clocks = <&gateclk 14>, <&gateclk 8>;
+	status = "disabled";
 };
 
 cpxor1@6c0000 {
@@ -212,6 +213,7 @@ cpxor1@6c0000 {
 	dma-coherent;
 	msi-parent = <&gic_v2m0>;
 	clocks = <&gateclk 14>, <&gateclk 7>;
+	status = "disabled";
 };
 
 thermal@400078 {
-- 
1.7.9.5

